### ISSUE TITLE
Agent HTTP Check misreporting warnings around SSL

### ISSUE THREAD

**Issue by miketheman** [DataDog/dd-agent#1999]
For a `http_check.yaml` file that looks like this:

```yaml
init_config:

instances:
- name: scores
  url: http://scores.example.com

- name: forum
  url: https://forum.example.com

- name: timeout
  url: https://timeout.example.com
```

Agent 5.4.6 (as well as 5.5.1) report:

```log
    http_check
    ----------
      - instance #0 [WARNING]
          Warning: Skipping SSL certificate validation for https://timeout.example.com based on configuration
          Warning: Using events for service checks is deprecated in favor of monitors and will be removed in future versions of the Datadog Agent.
          Warning: Using events for service checks is deprecated in favor of monitors and will be removed in future versions of the Datadog Agent.
          Warning: Using events for service checks is deprecated in favor of monitors and will be removed in future versions of the Datadog Agent.
          Warning: Using events for service checks is deprecated in favor of monitors and will be removed in future versions of the Datadog Agent.
          Warning: Using events for service checks is deprecated in favor of monitors and will be removed in future versions of the Datadog Agent.
      - instance #1 [OK]
      - instance #2 [WARNING]
          Warning: Skipping SSL certificate validation for https://forum.example.com based on configuration
      - Collected 3 metrics, 0 events & 6 service checks
```

All SSL Certs on web servers in config are valid - confirmed with both `curl` 7.35.0 default CA certs on Ubuntu 14.04 as well as with:

    curl -v --cacert /opt/datadog-agent/embedded/lib/python2.7/site-packages/requests/cacert.pem https://timeout.example.com

So I think there's a couple of issues:

- There seems to be some logic that triggers multiple "events for service check vs monitors" warnings, even thought there are no defined thresholds in the config.
- No user-defined certificate verification disabling appears to exist, why is it being done?

**Comment by miketheman** [comment]
This problem persists with Agent 5.6.0.

**Comment by yagni** [comment]
For whatever reason, Datadog disables SSL certificate validation by default. You have to manually tell it to do the validation for each instance in your check, like so:

```yaml
instances:
- name: scores
  url: http://scores.example.com
  disable_ssl_validation: true

- name: forum
  url: https://forum.example.com
  disable_ssl_validation: true

- name: timeout
  url: https://timeout.example.com
  disable_ssl_validation: true
```

That seemed to solve my "events for service checks" warnings as well.

**Comment by miketheman** [comment]
@yagni thanks for the heads-up - all I had to do was add the `disable_ssl_validation: true` to the one interface that isn't using SSL - the others have valid certs.

@Datadog/maintainers - this still constitutes a bug in my mind, and should result in better warning behavior, as well as a user should never need to disable ssl validation for an HTTP endpoint.

**Comment by yagni** [comment]
@miketheman I guess I'm a little confused (it doesn't help that the option name is in the negative). SSL validation should be disabled by default. Since that's equivalent to `disable_ssl_validation: true`, putting that in the yaml should be a noop. For HTTP endpoints, that seems like a reasonable default since they don't use SSL and your original report seems to confirm this--the HTTP endpoint doesn't have the SSL validation warnings.

But for SSL endpoints it spits out a warning if you leave the default because it's not a good idea to skip validating the certs in that case, which also makes sense to me. The warning should go away if you enable SSL validation via `disable_ssl_validation: false`. At least, it did in my case.

What doesn't make sense to me is why SSL validation is disabled by default for SSL endpoints--it seems like almost everyone would want that enabled by default.

Another source of confusion is that it's also not very well documented. In fact, it only appears in the example yaml within the conf.d directory. Seems to me it should be documented in http://docs.datadoghq.com/guides/network_checks/ as well.

**Comment by miketheman** [comment]
I'm still wrong. In Agent 5.6.1, disabling all still produces warnings.

**Comment by yannmh** [comment]
I agree, `disable_ssl_validation ` option, 'Skipping SSL certificate validation' and 'events for service checks' warnings are very confusing here. 

I am addding this card on the 5.7.0 milestone: we'll assess what can be done in order to makes things more intuitive and remove deprecated warnings.  Thanks for your valuable feedback @yagni @miketheman !

**Comment by remh** [comment]
Fixing this would be a compatibility breaking change.
Hence it will wait for version 6.0.0

**Comment by miketheman** [comment]
@remh @yannmh This issue is now over a year old, and is still throwing warnings for unrelated behavior and is confusing.
Is there a timeline for 6.0.0?

**Comment by spesnova** [comment]
any updates?

**Comment by rootd00d** [comment]
What we're noticing is that a _huge_ portion of our logs all look like this --

```
May 17 00:10:48 FOO agent[30484]: 2018-05-17 00:10:48 UTC | WARN | (datadog_agent.go:135 in LogMessage) | (__init__.py:319) | Skipping SSL certificate validation for https://localhost/auth/ based on configuration
May 17 00:10:48 FOO agent[30484]: /opt/datadog-agent/embedded/lib/python2.7/site-packages/urllib3/connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
May 17 00:10:48 FOO agent[30484]:   InsecureRequestWarning)
```

By huge I mean, they comprise >50% of the total log volume for several instances.  It's producing a lot of noise; and we'd like a way to turn it off.

Is there anyway this would be up for consideration?

**Comment by masci** [comment]
@rootd00d thanks for the report, going to demote the first log line to `DEBUG`, in the end if one explicitly sets `disable_ssl_validation` to True, we can assume they know what they're doing.

Currently investigating how to silence the other lines coming from `urllib3`, will keep this issue updated.

**Comment by calvinbui** [comment]
I've seeing these warnings on 7.16.1

**Comment by SomeUserForReport** [comment]
Still here Agent 7.57.2 - Commit: 38ba0c7 - Serialization version: v5.0.130 - Go version: go1.22.7
